core/src/test/scala/integration/kafka/api/BaseAsyncConsumerTest.scala

```diff
@@ -35,8 +39,12 @@ class BaseAsyncConsumerTest extends AbstractConsumerTest {
     consumer.commitAsync(cb)
     waitUntilTrue(() => {
       cb.successCount == 1
-    }, "wait until commit is completed successfully", 5000)
+    }, "wait until commit is completed successfully", defaultBlockingAPITimeoutMs)
+    val committedOffset = consumer.committed(Set(tp).asJava, Duration.ofMillis(defaultBlockingAPITimeoutMs))
+
     assertTrue(consumer.assignment.contains(tp))
+    assertNotNull(committedOffset)
+    assertNull(committedOffset.get(tp))
```

Comment:
Hmm, why would the committed offset be null for tp?

Reply:
In this test, we aren't committing any offset, so the actual return of the commit response looks like this:
(same for both old and the async one) `{topic-0=null}`

Reply:
Thanks, Philip. I still don't quite understand. The commitAsync/commitSync API commits the offset for all consumed partitions, right? This test assigns tp to the consumer.

Reply:
Right, but in the test there's no consumption in the test, i.e., we aren't polling the consumer, and therefore there isn't a valid position. So when commit is invoked w/o specifying an offset, it takes the consumed position, in the test case, there isn't any.

Reply:
Thanks for the explanation, Philip. I understand it now. Could you add the explanation as a comment in the code?

Reply:
Thank you @junrao - Sorry about the unclarity. Will add a comment to the test.

Reply:
Since this test is for no-consumption then the `sendRecords` could be removed in both tests I guess